azure-mgmt-monitor/src/main/java/com/microsoft/azure/management/monitor/MetricDefinitions.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.monitor;
+
+import com.microsoft.azure.management.monitor.implementation.MonitorManagementClientImpl;
+import com.microsoft.azure.management.monitor.implementation.MonitorManager;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.HasManager;
+import com.microsoft.azure.management.resources.fluentcore.model.HasInner;
+import rx.Observable;
+
+import java.util.List;
+
+
+/**
+ * Entry point for Monitor Metric Definitions API.
+ */
+public interface MetricDefinitions extends
+        HasManager<MonitorManager>,
+        HasInner<MonitorManagementClientImpl> {
+
+    /**
+     * Lists Metric Definitions for a given resource.
+     *
+     * @param resourceId The resource Id.
+     * @return list of metric definitions.
+     */
+    List<MetricDefinition> listByResource(String resourceId);
+
+    /**
+     * Lists Metric Definitions for a given resource.
+     *
+     * @param resourceId The resource Id.
+     * @return a representation of the deferred computation of Metric Definitions list call.
+     */
+    Observable<List<MetricDefinition>> listByResourceAsync(String resourceId);
```

Comment:
we should return an Observable<MetricDefinition> rather than the current. See:
    Observable<List<QueryKeyInner>> queryKeysList = this.manager().inner().queryKeys()
        .listBySearchServiceAsync(this.resourceGroupName(), this.name());

    return queryKeysList.flatMap(new Func1<List< QueryKeyInner>, Observable<QueryKeyInner>>() {
      @Override
      public Observable<QueryKeyInner> call(List<QueryKeyInner> queryKeyInners) {
        return Observable.from(queryKeyInners);
      }
    }).map(new Func1<QueryKeyInner, QueryKey>() {
      @Override
      public QueryKey call(QueryKeyInner queryKeyInner) {
        return new QueryKeyImpl(queryKeyInner);
      }
    });